core-web/apps/dotcms-ui/src/app/portlets/dot-containers/dot-container-create/dot-container-properties/dot-container-properties.component.ts

```diff
@@ -65,28 +65,47 @@ export class DotContainerPropertiesComponent implements OnInit {
                 }
             });
 
-        //
         this.form.valueChanges
             .pipe(takeUntil(this.destroy$), startWith(this.form.value), pairwise())
             .subscribe(([prevValue, currValue]) => {
                 this.store.updateFormStatus({
                     invalidForm: !this.form.valid,
                     container: currValue
                 });
-                if (
-                    (prevValue.maxContentlets === 0 || prevValue.maxContentlets === null) &&
-                    currValue.maxContentlets > 0
-                ) {
+                if (this.IsShowContentTypes(prevValue, currValue)) {
                     this.showContentTypeAndCode();
-                } else if (
-                    (prevValue.maxContentlets > 0 || prevValue.maxContentlets === null) &&
-                    currValue.maxContentlets === 0
-                ) {
+                } else if (this.IsShowClearConfirmationModal(prevValue, currValue)) {
                     this.clearContentConfirmationModal(prevValue.maxContentlets || 1);
                 }
             });
     }
 
+    /**
+     * check prev value and current value for showing the content type
+     * @param {*} prevValue
+     * @param {*} currValue
+     * @memberof DotContainerPropertiesComponent
+     */
```

Comment:
We should not have `any` (it's being inferred by typescript) here, we need types.

Reply:
Noted